django_refugeerights/snappyuploader/tasks.py

```diff
@@ -1,41 +1,200 @@
-from celery import task
-from django.conf import settings
+from __future__ import absolute_import
+
+from celery.task import Task
+from celery.utils.log import get_task_logger
+from celery.exceptions import SoftTimeLimitExceeded
 import requests
-from .models import SnappyFaq
 
+from django.conf import settings
+
+from .models import SnappyFaq, FailureLog
+
+import json
 
-def snappy_request(method, endpoint):
+logger = get_task_logger(__name__)
+
+
+def snappy_request(method, endpoint, py_data=None):
     """
     Function that makes the Snappy Api requests.
     """
     api_url = "%s/%s" % (settings.SNAPPY_BASE_URL, endpoint)
     auth = (settings.SNAPPY_API_KEY, 'x')
     headers = {'content-type': 'application/json; charset=utf-8'}
+    if method is "POST":
+        data = json.dumps(py_data)
+        response = requests.post(
+            api_url, auth=auth, data=data, headers=headers, verify=False)
+    elif method == "GET":
+        response = requests.get(
+            api_url, auth=auth, headers=headers, verify=False)
+    response.raise_for_status()
+    return response.json()
 
-    if method == "GET":
-        response = requests.get(api_url, auth=auth, headers=headers)
 
-    return response.json()
+def add_snappy_topic(topic_title, faq_id):
+    """
+    Function that makes a new Snappy topic via Api Post
+    """
+    data = {"topic": topic_title}
+    response = snappy_request(
+        "POST",
+        "account/%s/faqs/%s/topics" % (
+            settings.SNAPPY_ACCOUNT_ID, faq_id),
+        py_data=data)
+    # Set topic_id to new topic id
+    return response
+
+
+def is_valid_snappy_data(csv_entry):
+    is_valid = True
+    required_fields = ["lang", "topic", "question", "answer"]
+    for field in required_fields:
+        if not field in csv_entry:
```

Comment:
You can combine this into one if `if not field in csv_entry or csv_entry[field] == ""`